src/SecureConnection.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace React\Socket;
+
+/**
+ * A connection supporting transport layer security.
+ */
+class SecureConnection extends Connection
+{
+    protected $isSecure = false;
+    protected $protocolNumber;
+
+    public function handleData($stream)
+    {
+        if (! $this->isSecure) {
+            $enabled = stream_socket_enable_crypto($stream, true, $this->protocolNumber);
+            if ($enabled === false) {
+                $this
+                    ->err('Failed to complete a secure handshake with the client.')
+                    ->end()
+                ;
+                return;
+            } elseif ($enabled === 0) {
+                return;
+            }
+            $this->isSecure = true;
+            $this->emit('connection', array($this));
+        }
+
+        $data = fread($stream, $this->bufferSize);
+
+        if ('' !== $data && false !== $data) {
+            $this->emit('data', array($data, $this));
+        }
+
+        if (false === $data || !is_resource($stream) || feof($stream)) {
+            $this->end();
+        }
```

Comment:
Note that the first read after enabling crypto seems always to be empty: so ending on an empty read will prevent a subsequent non-empty read.